src/ws/cockpitauth.c

```diff
@@ -872,7 +886,7 @@ cockpit_auth_spawn_login_async (CockpitAuth *self,
 
       if (g_spawn_async_with_pipes (NULL, (gchar **) argv, NULL,
                                      G_SPAWN_DO_NOT_REAP_CHILD | G_SPAWN_LEAVE_DESCRIPTORS_OPEN,
-                                     spawn_child_setup, GINT_TO_POINTER (child_pd),
+                                     spawn_child_setup, child_fds,
```

Comment:
We're passing two very different things here in this array. One is an open file descriptor, the other is just a number. I would prefer a struct. Otherwise this code is hard to understand.